homeassistant/components/tahoma/sensor.py

```diff
@@ -0,0 +1,153 @@
+"""Support for TaHoma sensors."""
+import logging
+from typing import Optional
+
+from homeassistant.components.sensor import DOMAIN as SENSOR
+from homeassistant.const import (
+    CONCENTRATION_PARTS_PER_MILLION,
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_ILLUMINANCE,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_TEMPERATURE,
+    ELECTRICAL_CURRENT_AMPERE,
+    ENERGY_KILO_WATT_HOUR,
+    ENERGY_WATT_HOUR,
+    PERCENTAGE,
+    POWER_KILO_WATT,
+    POWER_WATT,
+    SPEED_METERS_PER_SECOND,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+    TEMP_KELVIN,
+    VOLT,
+    VOLUME_CUBIC_METERS,
+    VOLUME_LITERS,
+)
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+from .tahoma_entity import TahomaEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+CORE_CO2_CONCENTRATION_STATE = "core:CO2ConcentrationState"
+CORE_CO_CONCENTRATION_STATE = "core:COConcentrationState"
+CORE_ELECTRIC_ENERGY_CONSUMPTION_STATE = "core:ElectricEnergyConsumptionState"
+CORE_ELECTRIC_POWER_CONSUMPTION_STATE = "core:ElectricPowerConsumptionState"
+CORE_FOSSIL_ENERGY_CONSUMPTION_STATE = "core:FossilEnergyConsumptionState"
+CORE_GAS_CONSUMPTION_STATE = "core:GasConsumptionState"
+CORE_LUMINANCE_STATE = "core:LuminanceState"
+CORE_MEASURED_VALUE_TYPE = "core:MeasuredValueType"
+CORE_RELATIVE_HUMIDITY_STATE = "core:RelativeHumidityState"
+CORE_SUN_ENERGY_STATE = "core:SunEnergyState"
+CORE_TEMPERATURE_STATE = "core:TemperatureState"
+CORE_THERMAL_ENERGY_CONSUMPTION_STATE = "core:ThermalEnergyConsumptionState"
+CORE_WATER_CONSUMPTION_STATE = "core:WaterConsumptionState"
+CORE_WINDSPEED_STATE = "core:WindSpeedState"
+
+UNIT_LX = "lx"
+
+TAHOMA_SENSOR_DEVICE_CLASSES = {
+    "CO2Sensor": None,  # co2
```

Comment:
I will work on a `DEVICE_CLASS_CO2` proposal in architecture. Currently these are defined by HomeKit, but I guess I am not allowed to use them in this integration..

https://github.com/home-assistant/core/blob/5f9a1d105c32fd13d360e53de8e418b13c0c8f56/homeassistant/components/homekit/const.py#L239-L240

Reply:
https://github.com/home-assistant/architecture/discussions/520, started a discussion in architecture.

Reply:
Happy to add DEVICE_CLASS_CO2 from https://github.com/home-assistant/core/pull/47487, but first we would need to have `home-assistant:rework-tahoma` rebased on dev again.

Reply:
@bdraco could you perhaps rebase `home-assistant:rework-tahoma` when it suits you 👍🏻. I will refactor this PR and process all feedback.

Reply:
Fixed in b32d41b.